File: 264_UglyNumberII.cs
public class Solution {
    public int NthUglyNumber(int n) {
        if (n <= 0) return 0;
        List<int> uglies = new List<int>(){1};
        int i2 = 0, i3 = 0, i5 = 0; 
        
        while (uglies.Count < n) {
            int a = uglies[i2] * 2;
            int b = uglies[i3] * 3;
            int c = uglies[i5] * 5;
            
            int nextUgly = Math.Min(a, Math.Min(b,c));
            
            i2 = nextUgly == a? i2 + 1 : i2;
            i3 = nextUgly == b? i3 + 1 : i3;
            i5 = nextUgly == c? i5 + 1 : i5;
            
            uglies.Add(nextUgly);
        }
        return uglies[n-1];
    }
}

File: Python/264_UglyNumberII.py
class Solution:
    def nthUglyNumber(self, n: int) -> int:
        if n == 0: return 0
        i2, i3, i5 = 0,0,0
        ans = [1]
        for i in range(1,n):
            a = ans[i2] * 2
            b = ans[i3] * 3
            c = ans[i5] * 5
            
            m = min(a, b, c)
            
            i2 = i2 + 1 if m == a else i2
            i3 = i3 + 1 if m == b else i3
            i5 = i5 + 1 if m == c else i5
            ans.append(m)
        return ans[-1]
        